pkg/kubelet/kubelet_pods.go

```diff
@@ -1138,6 +1138,9 @@ func (kl *Kubelet) validateContainerLogStatus(podName string, podStatus *v1.PodS
 		cID = cStatus.ContainerID
 
 	case terminated != nil:
+		if terminated.ContainerID == "" {
+			return kubecontainer.ContainerID{}, fmt.Errorf("container %q in pod %q is terminated", containerName, podName)
+		}
```

Comment:
You could previously see logs of terminated containers.  Are you confident we only hit this case when the container was deleted just before we attempted to log from it, vs in general use?  (I assume so, just want to be triple sure)

Reply:
You were right... a container ID of the last container that actually was able to start was still available in `lastState`. I've updated it so that if `terminated` refers to a pod without a container ID (like in the case of a crash loop backoff scenario), we'll see if there's one in `lastState` that we can get instead.

Reply:
Excellent.